shtdlib.sh

```diff
@@ -204,17 +204,85 @@ function readlink_m {
     fi
 }
 
+# A function to get the status of shell options supported by 'set'
+# Outputs: 'on', 'off' or nothing for invalid/empty option names
+function get_sh_opts {
+    if [ -n "${1}" ]; then
+        shopt -o "${1}" 2> /dev/null | awk '{print $2}'
```

Comment:
wouldn't it be handier if this returned 0 or 1 and then just printed debug ?

Reply:
Hmm.. Yeah, that could work but I need a special case for an invalid option.. so I'll return 0 for 'on', 1 for 'off' and 1 for empty/invalid